src/main/java/nl/andrewl/emaildatasetbrowser/view/search/export/QueryExportParameterPanel.java

```diff
@@ -0,0 +1,117 @@
+package nl.andrewl.emaildatasetbrowser.view.search.export;
+
+import java.awt.FlowLayout;
+import java.nio.file.Path;
+import java.util.concurrent.CompletableFuture;
+
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JFileChooser;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JSpinner;
+import javax.swing.JTextField;
+import javax.swing.SpinnerNumberModel;
+import javax.swing.filechooser.FileNameExtensionFilter;
+
+import nl.andrewl.email_indexer.data.export.query.QueryExportParams;
+import nl.andrewl.email_indexer.data.export.query.QueryExporter;
+import nl.andrewl.emaildatasetbrowser.view.search.LuceneSearchPanel;
+
+/**
+ * Simple JPanel base class for Exporting queries.
+ */
+public abstract class QueryExportParameterPanel extends JPanel {
+    private LuceneSearchPanel searchPanel;
+
+    private final JSpinner maxResultsSpinner = new JSpinner(
+            new SpinnerNumberModel(100, 1, 10000, 1));
+    private final JCheckBox separateThreadsToggle = new JCheckBox("Separate mailing threads");
+    private final JTextField outputPathField = new JTextField();
```

Comment:
I think a lot of the complexity of this panel can be removed if you use `nl.andrewl.emaildatasetbrowser.view.PathSelectField`, which is essentially exactly what you're doing here, and supports customizing the file selection mode, whether to show all files, and file filters, and the label that's shown.